### ISSUE TITLE
Unhandled stream error in pipe

### ISSUE THREAD

**Issue by anuveyatsu** [ory/kratos-selfservice-ui-node#44]
**Describe the bug**

As described in #43 when piping request to response fails, the node app crashes due to unhandled error.

**To Reproduce**

Please refer to steps in #43 

**Expected behavior**

App doesn't crash and logs some messages and returns 500 error.

**Environment**

- Kratos-selfservice-ui-node: `v0.3.0-alpha.1`
- Kratos: `v0.3.0-alpha.1`
- Environment: quickstart (docker-compose)

**Additional context**

Logs:

```bash
kratos-selfservice-ui-node_1  | No request found in URL, initializing auth flow.
kratos-selfservice-ui-node_1  | GET /auth/login 302 148 - 42.718 ms
kratos_1                      | time="2020-05-22T12:35:39Z" level=info msg="started handling request" method=GET name="public#http://127.0.0.1:4455/.ory/kratos/public/" remote="172.20.0.4:49302" request=/self-service/browser/flows/login
kratos_1                      | time="2020-05-22T12:35:44Z" level=info msg="completed handling request" method=GET name="public#http://127.0.0.1:4455/.ory/kratos/public/" remote="172.20.0.4:49302" request=/self-service/browser/flows/login status=302 text_status=Found took=5.4979608s
kratos-selfservice-ui-node_1  | GET /.ory/kratos/public/self-service/browser/flows/login 302 100 - 5583.624 ms
kratos_1                      | time="2020-05-22T12:35:44Z" level=info msg="started handling request" method=GET name="admin#http://kratos:4434/" remote="172.20.0.4:46460" request="/self-service/browser/flows/requests/login?request=a1386f7e-b249-40a4-9fd9-38a87329ff8a"
kratos_1                      | time="2020-05-22T12:35:44Z" level=info msg="completed handling request" method=GET name="admin#http://kratos:4434/" remote="172.20.0.4:46460" request="/self-service/browser/flows/requests/login?request=a1386f7e-b249-40a4-9fd9-38a87329ff8a" status=200 text_status=OK took=1.9646ms
kratos-selfservice-ui-node_1  | GET /auth/login?request=a1386f7e-b249-40a4-9fd9-38a87329ff8a 200 13535 - 387.953 ms
kratos-selfservice-ui-node_1  | GET /index.css 304 - - 31.221 ms
kratos-selfservice-ui-node_1  | GET /typography.css 304 - - 35.093 ms
kratos-selfservice-ui-node_1  | GET /form.css 304 - - 25.768 ms
kratos-selfservice-ui-node_1  | GET /auth.css 304 - - 33.850 ms
kratos_1                      | time="2020-05-22T12:35:47Z" level=info msg="started handling request" method=POST name="public#http://127.0.0.1:4455/.ory/kratos/public/" remote="172.20.0.4:49314" request=/self-service/browser/flows/strategies/oidc/auth/a1386f7e-b249-40a4-9fd9-38a87329ff8a
kratos_1                      | time="2020-05-22T12:35:49Z" level=info msg="completed handling request" method=POST name="public#http://127.0.0.1:4455/.ory/kratos/public/" remote="172.20.0.4:49314" request=/self-service/browser/flows/strategies/oidc/auth/a1386f7e-b249-40a4-9fd9-38a87329ff8a status=302 text_status=Found took=1.7090371s
kratos-selfservice-ui-node_1  | POST /.ory/kratos/public/self-service/browser/flows/strategies/oidc/auth/a1386f7e-b249-40a4-9fd9-38a87329ff8a 302 0 - 1717.150 ms
kratos_1                      | time="2020-05-22T12:35:54Z" level=info msg="started handling request" method=GET name="public#http://127.0.0.1:4455/.ory/kratos/public/" remote="172.20.0.4:49320" request="/self-service/browser/flows/strategies/oidc/callback/github?code=...&state=..."
kratos_1                      | time="2020-05-22T12:35:59Z" level=debug msg="Received successful OpenID Connect callback but user is not registered. Re-initializing registration flow now." provider=github subject=...
kratos_1                      | time="2020-05-22T12:36:04Z" level=debug msg="OpenID Connect Jsonnet mapper completed." mapper_jsonnet_output="{\n   \"identity\": {\n      \"traits\": {\n         \"email\": \"your-email@mail.com\"\n      }\n   }\n}\n" mapper_jsonnet_url="file:///etc/config/kratos/oidc.github.jsonnet" oidc_claims="&{https://github.com/login/oauth/access_token ... Anuar Ustayev       https://github.com/anuveyatsu https://avatars1.githubusercontent.com/u/...?v=4  your-email@mail.com true      false 1590150887}" oidc_provider=github
kratos_1                      | time="2020-05-22T12:36:06Z" level=debug msg="A new identity has registered using self-service registration. Running post execution hooks." identity_id=d39fe51c-e8e4-4f8b-919e-0f3bdd26a45c
kratos_1                      | time="2020-05-22T12:36:07Z" level=debug msg="Post registration execution hooks completed successfully." identity_id=d39fe51c-e8e4-4f8b-919e-0f3bdd26a45c
kratos_1                      | time="2020-05-22T12:36:07Z" level=info msg="completed handling request" method=GET name="public#http://127.0.0.1:4455/.ory/kratos/public/" remote="172.20.0.4:49320" request="/self-service/browser/flows/strategies/oidc/callback/github?code=...&state=... status=302 text_status=Found took=13.8210832s
kratos-selfservice-ui-node_1  | internal/streams/legacy.js:61
kratos-selfservice-ui-node_1  |       throw er; // Unhandled stream error in pipe.
kratos-selfservice-ui-node_1  |       ^
kratos-selfservice-ui-node_1  | 
kratos-selfservice-ui-node_1  | Error: socket hang up
kratos-selfservice-ui-node_1  |     at connResetException (internal/errors.js:609:14)
kratos-selfservice-ui-node_1  |     at Socket.socketOnEnd (_http_client.js:455:23)
kratos-selfservice-ui-node_1  |     at Socket.emit (events.js:333:22)
kratos-selfservice-ui-node_1  |     at endReadableNT (_stream_readable.js:1220:12)
kratos-selfservice-ui-node_1  |     at processTicksAndRejections (internal/process/task_queues.js:84:21) {
kratos-selfservice-ui-node_1  |   code: 'ECONNRESET'
kratos-selfservice-ui-node_1  | }
kratos-selfservice-ui-node_1  | npm ERR! code ELIFECYCLE
kratos-selfservice-ui-node_1  | npm ERR! errno 1
kratos-selfservice-ui-node_1  | npm ERR! kratos-selfservice-ui-node@0.0.1 serve: `node lib/index.js`
kratos-selfservice-ui-node_1  | npm ERR! Exit status 1
kratos-selfservice-ui-node_1  | npm ERR! 
kratos-selfservice-ui-node_1  | npm ERR! Failed at the kratos-selfservice-ui-node@0.0.1 serve script.
kratos-selfservice-ui-node_1  | npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
kratos-selfservice-ui-node_1  | 
kratos-selfservice-ui-node_1  | npm ERR! A complete log of this run can be found in:
kratos-selfservice-ui-node_1  | npm ERR!     /root/.npm/_logs/2020-05-22T12_36_07_942Z-debug.log
kratos_kratos-selfservice-ui-node_1 exited with code 1
```

**Comment by aeneasr** [comment]
Interesting, would you be up for a PR? :)

**Comment by anuveyatsu** [comment]
@aeneasr sure 😄

**Comment by anuveyatsu** [comment]
FIXED in #45
